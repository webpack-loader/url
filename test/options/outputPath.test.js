const { webpack } = require('@webpack-utilities/test')

describe('Options', () => {
  describe('outputPath', () => {
    test('{String}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath: 'output_path/'
          }
        }
      }

      const stats = await webpack('fixture.js', config)
      const { assets, source } = stats.toJson().modules[0]

      expect({ assets, source }).toMatchSnapshot()
    })

    test('{Function}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            outputPath(url) {
              return `output_path_fn/${url}`
            }
          }
        }
      }

      const stats = await webpack('fixture.js', config)
      const { assets, source } = stats.toJson().modules[0]

      expect({ assets, source }).toMatchSnapshot()
    })
  })
})
