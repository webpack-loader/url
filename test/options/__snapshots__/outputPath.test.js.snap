// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options outputPath {Function} 1`] = `
Object {
  "assets": Array [
    "output_path_fn/39f5c21c.png",
  ],
  "source": "export default __webpack_public_path__ + \\"output_path_fn/39f5c21c.png\\";",
}
`;

exports[`Options outputPath {String} 1`] = `
Object {
  "assets": Array [
    "output_path/39f5c21c.png",
  ],
  "source": "export default __webpack_public_path__ + \\"output_path/39f5c21c.png\\";",
}
`;
